src/libraries/Microsoft.Extensions.Options/gen/Parser.cs

```diff
@@ -467,7 +470,81 @@ private List<ValidatedMember> GetMembersToValidate(ITypeSymbol modelType, bool s
+                            attributeFullQualifiedName = $"{Emitter.StaticGeneratedValidationAttributesClassesNamespace}.{Emitter.StaticAttributeClassNamePrefix}{_commonUtilities.Suffix}_{_symbolHolder.CompareAttributeSymbol.Name}";
+                        }
+                    }
+                    else if (SymbolEqualityComparer.Default.Equals(attributeType, _symbolHolder.RangeAttributeSymbol))
+                    {
+                        ImmutableArray<IParameterSymbol> constructorParameters = attribute.AttributeConstructor?.Parameters ?? ImmutableArray<IParameterSymbol>.Empty;
+                        SpecialType argumentSpecialType = SpecialType.None;
+                        if (constructorParameters.Length == 2)
+                        {
+                            argumentSpecialType = constructorParameters[0].Type.SpecialType;
+                        }
+                        else if (constructorParameters.Length == 3)
+                        {
+                            object? argumentValue = null;
+                            for (int i = 0; i < constructorParameters.Length; i++)
+                            {
+                                if (constructorParameters[i].Name == "type")
+                                {
+                                    argumentValue = attribute.ConstructorArguments[i].Value;
+                                    break;
+                                }
+                            }
+
+                            if (argumentValue is INamedTypeSymbol namedTypeSymbol && CommonUtilities.IsConvertibleBasicType(namedTypeSymbol))
+                            {
+                                argumentSpecialType = namedTypeSymbol.SpecialType;
+                            }
+                        }
+
+                        ITypeSymbol typeSymbol = memberType;
+                        if (typeSymbol.OriginalDefinition.SpecialType == SpecialType.System_Nullable_T)
+                        {
+                            typeSymbol = ((INamedTypeSymbol)typeSymbol).TypeArguments[0];
+                        }
+
+                        if (argumentSpecialType != SpecialType.None &&
+                            CommonUtilities.IsConvertibleBasicType(typeSymbol) &&
+                            (constructorParameters.Length != 3 || typeSymbol.SpecialType == argumentSpecialType)) // When type is provided as a parameter, it has to match the property type.
+                        {
+                            _commonUtilities.EnsureTrackingAttribute(_symbolHolder.RangeAttributeSymbol.Name, createValue: false, out HashSet<object>? trackedTypeList);
+                            attributeFullQualifiedName = $"{Emitter.StaticGeneratedValidationAttributesClassesNamespace}.{Emitter.StaticAttributeClassNamePrefix}{_commonUtilities.Suffix}_{_symbolHolder.RangeAttributeSymbol.Name}";
+                        }
```

Comment:
Suggestion: Refactor this out to a `TrackRangeAttributeIfNeeded` (or similar) helper method that contains the logic for tracking the attribute if needed.